### PR TITLE
Initial prototype for 202 template types

### PR DESCRIPTION
**Pull request by chfw**
first of all, integration tests has not been done yet. but because this is a team work, I would obey the principle of integrate early and commit per day. Also, in this way, @ayan-b could kick-start #234 on top of what's going here.

### CODE REVIEW

**Comment by chfw** [comment]
@ayan-b, didn't notice your comments till now. will look!

File: tests/fixtures/mobanengine/sample_template_type.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+template_types:
+  custom_jinja:
+      base_type: jinja2  # use base_type, instead of overrides
+      file_extensions:
+        - moban
+        - new
+        - demo_file_suffix
+      options:
+        extensions:
+          - jinja2.ext.do
```
Comment (ayan-b) [inline tests/fixtures/mobanengine/sample_template_type.yml]: EOL at EOF is missing.

File: moban/mobanfile/__init__.py [hunk 2]
```diff
@@ -62,6 +62,11 @@ def handle_moban_file_v1(moban_file_configurations, command_line_options):
     if extensions:
         plugins.ENGINES.register_extensions(extensions)
 
+    template_types = moban_file_configurations.get(
+        constants.LABEL_TEMPLATE_TYPES)
+    if template_types:
+        plugins.ENINGES.register_options(template_types)
```
Comment (ayan-b) [inline moban/mobanfile/__init__.py]: I think this should be `ENGINES`.

**Comment by chfw** [comment]
@ayan-b have addressed your feedback now.

**Comment by chfw** [comment]
more unit tests will be written along this feature development. but taking a cut today won't have sufficient unit tests.

**Pull request merged**